File: app/src/main/java/br/com/algorit/mangaalert/roomdatabase/database/AlertRoomDatabase.java
package br.com.algorit.mangaalert.database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import br.com.algorit.mangaalert.dao.NovelDao;
import br.com.algorit.mangaalert.model.Novel;

@Database(entities = {Novel.class}, version = 1, exportSchema = false)
public abstract class AlertRoomDatabase extends RoomDatabase {

    private static AlertRoomDatabase INSTANCE;

    public static AlertRoomDatabase getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (AlertRoomDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            AlertRoomDatabase.class, "MANGA_ALERT_DATABASE")
                            .build();
                }
            }
        }
        return INSTANCE;
    }

    public abstract NovelDao novelDao();
}

File: app/src/main/java/br/com/algorit/mangaalert/roomdatabase/database/AlertRoomDatabase.kt
package br.com.algorit.mangaalert.roomdatabase.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import br.com.algorit.mangaalert.roomdatabase.dao.MangaDao
import br.com.algorit.mangaalert.roomdatabase.dao.NovelDao
import br.com.algorit.mangaalert.roomdatabase.model.Manga
import br.com.algorit.mangaalert.roomdatabase.model.Novel

@Database(entities = [Manga::class, Novel::class], version = 1, exportSchema = false)
abstract class AlertRoomDatabase : RoomDatabase() {
    abstract fun novelDao(): NovelDao
    abstract fun mangaDao(): MangaDao

    companion object {
        private var INSTANCE: AlertRoomDatabase? = null
        @JvmStatic
        fun getDatabase(context: Context): AlertRoomDatabase? {
            if (INSTANCE == null) {
                synchronized(AlertRoomDatabase::class.java) {
                    if (INSTANCE == null) {
                        INSTANCE = Room.databaseBuilder(context.applicationContext,
                                AlertRoomDatabase::class.java, "MANGA_ALERT_DATABASE")
                                .build()
                    }
                }
            }
            return INSTANCE
        }
    }
}